tests/serverpod_test_server/test_integration/database_operations/crud/create_test.dart

```diff
@@ -43,7 +43,8 @@ void main() async {
 
       expect(
         UniqueData.db.insert(session, data),
-        throwsA(isA<DatabaseException>()),
+        throwsA(isA<DatabaseQueryException>()
+            .having((e) => e.code, 'code', '23505' /* Unique violation */)),
```

Comment:
Question: Would it make sense to have known [PG exception codes ](https://www.postgresql.org/docs/current/errcodes-appendix.html) defined in an abstract final class?

Reply:
Yes, it does... I took a shortcut here.

Let me introduce them as statics.

Reply:
This would be super! Suggestion to see if it makes sense to enumerate some categories as well (for example constraint violation). This is just a thought, not sure if it is worth while!

Reply:
They don't seem to be separated in error code sense, so I opted for representing the table as it is.

Since this boils down to a string comparison that is different for MySQL, I opted to keep the solution simple.

Reply:
The PG codes are hierarchical, with the first 2 digits being the class, e.g:
Class 23 — Integrity Constraint Violation

These classes could conceivably be maintained across DBs to some extent. But not essential right now! 👍

Reply:
Aah yes, that is correct! We will opt for the simple solution for now.